src/helpers/block_requests.rs

```diff
@@ -98,102 +98,124 @@ pub fn find_common_ancestor<N: Network>(canon: &LedgerState<N>, block_locators:
     Ok((maximum_common_ancestor, first_deviating_locator))
 }
 
+/// A case annotation enum for the block request handler.
```

Comment:
suggestion: perhaps this enum could be the place to document all the cases (as doc comments for each variant), as opposed to `handle_block_requests`?

Reply:
Moving it over now

Reply:
Done